File: runtime-go/internal/encoder/jsonl.go
package encoder

import (
	"encoding/json"

	"github.com/linxinhong/eventweave/runtime-go/internal/event"
)

// JSONL returns the canonical JSON representation of an event.
type JSONL struct{}

func init() {
	Register("jsonl", JSONL{})
}

// Name returns the encoder name.
func (JSONL) Name() string { return "jsonl" }

// ContentType returns the MIME type of the encoded output.
func (JSONL) ContentType() string { return "application/x-ndjson" }

// Encode marshals the event to JSON.
func (JSONL) Encode(ev event.Event) ([]byte, error) {
	return json.Marshal(ev)
}

File: eventweave/encoders/jsonl.py
"""Canonical JSONL encoder."""

from __future__ import annotations

import json

from eventweave.core.event import Event
from eventweave.encoders.base import Encoder, EncodeResult
from eventweave.encoders.registry import encoder


@encoder("jsonl", content_type="application/x-ndjson")
class JsonlEncoder(Encoder):
    """Encode an event as a single line of JSON."""

    name = "jsonl"
    content_type = "application/x-ndjson"
    description = "Canonical EventWeave JSON Lines encoder."

    def encode(self, event: Event) -> EncodeResult:
        return EncodeResult(
            success=True,
            output=json.dumps(event.model_dump(), default=str, ensure_ascii=False),
        )
